services/workflows-service/src/collection-flow/controllers/collection-flow.signup.controller.intg.test.ts

```diff
@@ -0,0 +1,176 @@
+        CollectionFlowService,
+        WorkflowTokenService,
+        WorkflowDefinitionRepository,
+        WorkflowRuntimeDataRepository,
+        CustomerRepository,
+        EndUserRepository,
+      ],
+    }).compile();
+
+    app = module.createNestApplication();
+    await app.init();
+
+    prismaClient = module.get<PrismaService>(PrismaService);
+    workflowTokenService = module.get<WorkflowTokenService>(WorkflowTokenService);
+    workflowDefinitionRepository = module.get<WorkflowDefinitionRepository>(
+      WorkflowDefinitionRepository,
+    );
+    workflowRuntimeDataRepository = module.get<WorkflowRuntimeDataRepository>(
+      WorkflowRuntimeDataRepository,
+    );
+    customerRepository = module.get<CustomerRepository>(CustomerRepository);
+    endUserRepository = module.get<EndUserRepository>(EndUserRepository);
+  });
+
+  beforeEach(async () => {
+    await cleanupDatabase();
+
+    const customer = await customerRepository.create({
+      data: {
+        name: 'signup-test-customer',
+        displayName: 'Signup Test Customer',
+        logoImageUri: 'test',
+      },
+    });
+
+    project = await createProject(prismaClient, customer, 'signup-test-project');
+
+    const workflowDefinition = await workflowDefinitionRepository.create({
+      data: {
+        name: 'signup-test-definition',
+        projectId: project.id,
+        definitionType: 'collectionFlow',
+        definition: {},
+      },
+    });
+
+    const { id: workflowRuntimeDataId } = await workflowRuntimeDataRepository.create({
+      data: {
+        workflowDefinitionId: workflowDefinition.id,
+        projectId: project.id,
+        workflowDefinitionVersion: 1,
+        context: {},
+      },
+    });
+
+    const now = new Date();
+    const expiresAt = new Date(now.setDate(now.getDate() + 7));
+
+    workflowRuntimeDataToken = await workflowTokenService.create(project.id, {
+      workflowRuntimeDataId,
+      expiresAt,
+    });
+  });
+
+  afterAll(async () => {
+    await tearDownDatabase();
+    await app.close();
+  });
+
+  describe('POST /collection-flow/signup', () => {
+    it('should create a new EndUser and attach it to the WorkflowRuntimeDataToken', async () => {
+      const signupDto = {
+        firstName: 'John',
+        lastName: 'Doe',
+        email: 'email@email.com',
+      };
+
+      expect(workflowRuntimeDataToken.endUserId).toBeNull();
+
+      const response = await request(app.getHttpServer())
+        .post('/collection-flow/signup')
+        .send(signupDto)
+        .set('authorization', `Bearer ${workflowRuntimeDataToken.token}`);
+
+      expect(response.status).toBe(201);
+
+      const workflowToken = await workflowTokenService.findByToken(workflowRuntimeDataToken.token);
+      expect(workflowToken?.endUserId).toBeDefined();
+
+      const endUser = await endUserRepository.findById(workflowToken?.endUserId ?? '', {}, [
+        project.id,
+      ]);
+      expect(endUser).toBeDefined();
+    });
```

Comment:
Create a test with 2 different end users and try to connect a non related end user, currently the test will pass...